mintapi/api.py

```diff
@@ -459,6 +472,87 @@ def __x_months_ago(self, months=2):
             day=1
         )
 
+    def get_trend_data(
+        self,
+        report_type: ReportView.Options,
+        date_filter: DateFilter.Options,
+        start_date: Optional[str] = None,
+        end_date: Optional[str] = None,
+        category_ids: List[str] = None,
+        tag_ids: List[str] = None,
+        descriptions: List[str] = None,
+        match_all_filters: bool = True,
+        limit: int = 1000,
+        offset: int = 0,
+    ) -> List[Dict]:
+        """
+        Public accessor for trend data. Internally constructs a trend api payload
+
+
+        Parameters
+        ----------
+        report_type : ReportView.Options
+            type of report to generate. must use one of enum values
+        date_filter : DateFilter.Options
+            date window. must use predefined enum windows or pass a start and end date with CUSTOM enum
+        start_date : Optional[str], optional
+            optional start date (YYYY-mm-dd) if using enum CUSTOM, by default None
+        end_date : Optional[str], optional
+            optional end date (YYYY-mm-dd) if using enum CUSTOM, by default None
+        category_ids : List[str], optional
+            optional list of category ids to filter by, by default None
+        tag_ids : List[str], optional
+            optional list of tag ids to filter by, by default None
+        descriptions : List[str], optional
+            optional list of descriptions (ui labeled merchants) to filter by, by default None
+        match_all_filters : bool, optional
+            whether to match all (True) supplied filters or any (False), by default True
+        limit : int, optional
+            page size, by default 1000
+        offset : int, optional
+            offset pagination for next pages, by default 0
+
+        Returns
+        -------
+        List[Dict]
+            returns a list of trend results (each dict)
+        """
+        search_clauses = []
+        if category_ids:
```

Comment:
I'd like an individual method that at the very least handles building the search_clauses.  Ideally, we have methods broken up into each type.